File: baekjoon/7000/7570.cpp
#include <bits/stdc++.h>
using namespace std;

const int max_N = 1000000;

int con[max_N + 1];

int main() {
    cin.tie(NULL), cout.tie(NULL);
    ios::sync_with_stdio(false);

    int N; cin >> N;
    int ans = 0;
    for (int i = 0; i < N; ++i) {
        int h; cin >> h;
        if (con[h - 1]) {
            con[h] = con[h - 1] + 1;
            ans = max(ans, con[h]);
        }
        else con[h] = 1;
    }
    cout << N - ans;
    return 0;
}

File: baekjoon/7000/7570.rs
use std::io::*;

struct Scanner<'a> {
    it: std::str::SplitWhitespace<'a>,
}

impl<'a> Scanner<'a> {
    fn new(s: &'a str) -> Scanner<'a> {
        Scanner {
            it: s.split_whitespace(),
        }
    }
    fn read<T: std::str::FromStr>(&mut self) -> T {
        self.it.next().unwrap().parse::<T>().ok().unwrap()
    }
}

fn main() {
    let mut s = String::new();
    stdin().read_to_string(&mut s).unwrap();
    let mut sc = Scanner::new(&s);
    let out = stdout();
    let mut out = BufWriter::new(out.lock());
    run(&mut sc, &mut out);
}

fn run<W: Write>(sc: &mut Scanner, out: &mut BufWriter<W>) {
    let n = sc.read::<usize>();
    let mut con = vec![0; n + 1];
    for _ in 0..n {
        let i = sc.read::<usize>();
        con[i] = if con[i - 1] == 0 { 1 } else { con[i - 1] + 1 };
    }
    let ans = n - con.into_iter().max().unwrap() as usize;
    writeln!(out, "{ans}").ok();
}
